### PR TITLE
Added check if a connected graph is bipartite or not

### PR DESCRIPTION
**Pull request by Ritish758**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

Check if a connected graph is bipartite or not #2680 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2680

### CODE REVIEW

File: C-Plus-Plus/graphs/check_if_graph_is_bipartite.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+is bipartite or not. For that firstly we need to
+Know what is a bipartite graph.
+Graph which is 2-colorable is a bipartite means if
+we can color the nodes of graph with just 2 colors
+where every 2 nodes joined by a edge have different
+colors
+We can solve this problem by various methods like
+BFS,DFS,Backtracking.
+Also I am taking unweighted and undirected and connected graph
+I am showing this problem by use of DFS and adjacency
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/check_if_graph_is_bipartite.cpp:13]: Use a more standard description

File: C-Plus-Plus/graphs/check_if_graph_is_bipartite.cpp [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            visited[u]=true;
+            // providing color to node
+            color[u]=1-color[n];
+            // traversing the edges of node u
+            if(!isbipartite(adj,u,visited,color)){
+                    return false;
+            }
+        }
+        // if node is visited then it must have different color with n
+        // else graph is not bipartite
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/check_if_graph_is_bipartite.cpp:38]: Prefer multi-line comments

File: C-Plus-Plus/graphs/check_if_graph_is_bipartite.cpp [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+Output
+Graph is bipartite
+Color of node 1 is: 0
+Color of node 2 is: 1
+Color of node 3 is: 0
+Color of node 4 is: 1
+Color of node 5 is: 0
+Color of node 6 is: 1
+
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/check_if_graph_is_bipartite.cpp:103]: You didn't even bother to complete the comment

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Make the README changes as well

**Comment by gargVader** [comment]
Duplicate #1510

**Pull request closed without merging**